### PR TITLE
Enable test_cast_matmul.py for XPU.

### PR DESCRIPTION
**Pull request by ienkovich**

### CODE REVIEW

File: python/test/regression/test_cast_matmul.py [hunk 1]
```diff
@@ -97,4 +102,10 @@ def matmul_kernel(A, B, C, M, N, K,  #
         BLOCK_N=BLOCK_N,  #
         BLOCK_K=BLOCK_K)
 
-    torch.testing.assert_close(out_torch, out_triton, atol=0.3, rtol=0.01)
+    # For XPU tests torch can compute reference result on CPU using fp32
+    # arithmetics for fp16 test. Such reference requires additionalincreased
```
Comment (whitneywhtsang) [inline python/test/regression/test_cast_matmul.py:106]:
```suggestion
    # arithmetics for fp16 test. Such reference requires additional increased
```

**Comment by ienkovich** [comment]
This enables one of the regression tests. This one tests matmul with casts. The original test fails for FP16 output type on big sizes. This happens because the reference result is computed on CPU with FP32 arithmetic and this causes significant differences compared to FP16 arithmetic. I chose to increase tolerance for this case. Alternatively, we can skip it or compute reference results differently (I tried computing via nested loop forcing conversion back to FP16 after each FMA, this works but takes too much time).

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -183,6 +183,11 @@ jobs:
           cd python/test/unit
           python3 -m pytest -n 8 --verbose operators
 
+      - name: Run partial regression tests
+        run: |
+          cd python/test
+          python3 -m pytest -n 8 --verbose regression/ --ignore=regression/test_performance.py --ignore=regression/test_functional_regressions.py
```
Comment (whitneywhtsang) [inline .github/workflows/build_and_test.yml:189]: To match integreation-tests.yml
```suggestion
      - name: Regression tests
        run: |
          cd python/test/regression
          python3 -m pytest -vvv -s . --rerun 10 --ignore=test_performance.py --ignore=test_functional_regressions.py
```

File: python/test/regression/test_cast_matmul.py (hunk 1 above)
Comment (ienkovich) [reply python/test/regression/test_cast_matmul.py]: Fixed

File: .github/workflows/build_and_test.yml (hunk 2 above)
Comment (ienkovich) [reply .github/workflows/build_and_test.yml]: Fixed

**Review by whitneywhtsang — APPROVED** [review]
We should revisit all increased tolerance once DPAS is implemented.

**Comment by ienkovich** [comment]
> We should revisit all increased tolerance once DPAS is implemented.

I don't think DPAS is going to help here. Increased tolerance is caused by the reference results computation, not the Triton one.

**Comment by whitneywhtsang** [comment]
> > We should revisit all increased tolerance once DPAS is implemented.
> 
> I don't think DPAS is going to help here. Increased tolerance is caused by the reference results computation, not the Triton one.

According to https://github.com/intel/intel-xpu-backend-for-triton/pull/360#discussion_r1467724750, DPAS instructions rather than FMAs could provide higher precision, but may not be enough for this case where the reference result is computed on CPU with FP32 arithmetic.

**Comment by ienkovich** [comment]
> According to [#360 (comment)](https://github.com/intel/intel-xpu-backend-for-triton/pull/360#discussion_r1467724750), DPAS instructions rather than FMAs could provide higher precision, but may not be enough for this case where the reference result is computed on CPU with FP32 arithmetic.

Well, we will have a chance to check this in practice. But note, that this case is different from the referenced tutorial. In the tutorial, matmul uses FP32 accumulator and casts it to FP16 only at the end of the computation, so it better matches what we have on CPU. Here accumulator is always FP16.

File: python/test/regression/test_cast_matmul.py [hunk 3]
```diff
@@ -97,4 +101,10 @@ def matmul_kernel(A, B, C, M, N, K,  #
         BLOCK_N=BLOCK_N,  #
         BLOCK_K=BLOCK_K)
 
-    torch.testing.assert_close(out_torch, out_triton, atol=0.3, rtol=0.01)
+    # For XPU tests torch can compute reference result on CPU using fp32
+    # arithmetics for fp16 test. Such reference requires increased
+    # tolerance for big K values.
+    if device == "xpu" and out_dtype == "float16" and K > 128:
```
Comment (etiotto) [inline python/test/regression/test_cast_matmul.py:107]: Add a # FIXME marker so that is easy to grep for tests that need to be fixed in the future.

**Review by etiotto — APPROVED** [review]

File: .github/workflows/build_and_test.yml [hunk 4]
```diff
@@ -183,6 +183,12 @@ jobs:
           cd python/test/unit
           python3 -m pytest -n 8 --verbose operators
 
+      - name: Regression tests
```
[the remark this answers is not in the document]
Comment (ienkovich) [reply .github/workflows/build_and_test.yml:186]: Added

File: python/test/regression/test_cast_matmul.py [hunk 5]
```diff
@@ -7,10 +7,14 @@
...
 import pytest
 import torch
+import intel_extension_for_pytorch  # type: ignore # noqa: F401
 
 import triton.language as tl
 from triton import cdiv, jit
 
+# FIXME remove this once Triton L0 queue and IPEX SYCL queue can be synchronized through events
+torch.xpu.enable_sync_mode()
+
```
[the remark this answers is not in the document]
Comment (ienkovich) [reply python/test/regression/test_cast_matmul.py]: Removed

File: python/test/regression/test_cast_matmul.py (hunk 3 above)
Comment (ienkovich) [reply python/test/regression/test_cast_matmul.py:104]: Added

**Pull request merged**